### PR TITLE
Update `HelmRelease` API to v2 (GA)

### PR DESCRIPTION
**Pull request by stefanprodan**
Changes:
- Update helm-controller to v1.0.0
- Update CLI commands to work with HelmRelease v2
- Update integration tests to HelmRelease v2
- Mark HelmRelease commands as GA

Part of: #4712

### CODE REVIEW

File: cmd/flux/testdata/trace/helmrelease-oci.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ kind: Namespace
 metadata:
   name: {{ .ns }}
 ---
-apiVersion: helm.toolkit.fluxcd.io/v2beta2
+apiVersion: helm.toolkit.fluxcd.io/v2
```
Comment (souleb) [inline cmd/flux/testdata/trace/helmrelease-oci.yaml:12]: the `kustomization` in this file also contains `validation: client`

File: cmd/flux/testdata/trace/helmrelease.yaml [hunk 2]
```diff
@@ -9,7 +9,7 @@ kind: Namespace
 metadata:
   name: {{ .ns }}
 ---
-apiVersion: helm.toolkit.fluxcd.io/v2beta2
+apiVersion: helm.toolkit.fluxcd.io/v2
```
Comment (souleb) [inline cmd/flux/testdata/trace/helmrelease.yaml:12]: the `kustomization` in this file also contains `validation: client`

File: cmd/flux/create_helmrelease.go [hunk 3]
```diff
@@ -39,14 +33,19 @@ import (
...
+	"github.com/fluxcd/flux2/v2/internal/flags"
+	"github.com/fluxcd/flux2/v2/internal/utils"
 )
 
 var createHelmReleaseCmd = &cobra.Command{
 	Use:     "helmrelease [name]",
 	Aliases: []string{"hr"},
 	Short:   "Create or update a HelmRelease resource",
-	Long:    withPreviewNote(`The helmrelease create command generates a HelmRelease resource for a given HelmRepository source.`),
+	Long:    `The helmrelease create command generates a HelmRelease resource for a given HelmRepository source.`,
```
[the remark this answers is not in the document]
Comment (stefanprodan) [reply cmd/flux/create_helmrelease.go:48]: This will be addressed in https://github.com/fluxcd/flux2/pull/4766

**Review by souleb — APPROVED** [review]
LGTM!

**Pull request merged**